src/routes/safe/components/Apps/utils.js

```diff
@@ -15,22 +18,44 @@ export const getAppInfoFromOrigin = (origin: string) => {
 }
 
 export const getAppInfoFromUrl = async (appUrl: string) => {
+  let res = { id: undefined, url: appUrl, name: 'unknown', iconUrl: appsIconSvg, error: true }
+
   try {
     const appInfo = await axios.get(`${appUrl}/manifest.json`)
-    const res = { url: appUrl, ...appInfo.data, iconUrl: appsIconSvg }
+
+    // verify imported app fulfil safe requirements
+    if (
+      !appInfo ||
+      !appInfo.data ||
+      !appInfo.data.name ||
+      !appInfo.data.description ||
+      !appInfo.data.providedBy ||
+      !appInfo.data.providedBy.name ||
+      !appInfo.data.providedBy.url
```

Comment:
Does it make sense that the developer can set this? shouldn't this url always be the url of the app. Just asking because this is extremely easy to spoof.

Reply:
providedBy is disabled for now. won't be required